common-rest/src/main/java/org/uniprot/api/rest/controller/BasicSearchController.java

```diff
@@ -279,6 +286,44 @@ protected MessageConverterContext<T> createStreamContext(
         return context;
     }
 
+    protected ResponseEntity<JobStatusResponse> createAsyncDownloadStatus(
+            DownloadJob job, String url) {
```

Comment:
Here, in this method parameter 'url' has not been used.

Also, I doubt whether this class (BasicSearchController) is the best place to keep this method. I would prefer to have it in a child class s.t. like 'AsyncDownloadableSeachController' or keep it a helper class.

Minor comment: I prefer a name like getAsyncDownloadStatus since it is just wrapping the already created status.